src/views/v2/widgets/widget-events-list.php

```diff
@@ -45,8 +46,9 @@
 	<div class="tribe-common-l-container">
 		<div class="tribe-events-widget-events-list">
 
-			<?php $this->template( 'components/json-ld-data' ); ?>
-
+			<?php if ( ! empty( $jsonld_enable ) ) : ?>
+				<?php $this->template( 'components/json-ld-data' ); ?>
+			<?php endif; ?>
```

Comment:
Maybe @lucatume has another opinion on this, but I personally don't think `$jsonld_enable` is something we need in the FE of the views. If `$jsonld_enable` is `false`, `$json_ld_data` is a blank string. If `$jsonld_enable` is `true`, then `$json_ld_data` is untouched.

Reply:
I think the fix idea, here, would be not to print the JSON-LD data container at all, if the JSON-LD is not enabled.
@Camwyn would it make sense to remove the check from this code and move it into the `components/json-ld-data` one? There the whole template would not be printed at all if there is nothing to print. Which would keep making sense in the case where JSON-LD is enabled, but there is nothing to print.

Reply:
The `components/json-ld-data` template is used in all the views as well as this one, so doing that wouldn't work so well.

Reply:
Here's the issues:

Paul:
- `$jsonld_enable` is sometime null, not false - thus the `empty`. Also, if it's null or false, the value of `$jason_ld_data` is NOT blank: http://p.tri.be/BWamnv
- I can change this - just letting you know what the current situation is - as the comments don't reflect current code. 

Luca:
- It's a shared template - the main views (month, list, etc) use it too - so adding in a check for something that they don't necessarily pass seems suspect? This lends more weight to blanking out the `$json_ld_data` if `$jsonld_enable` is false.
- "...not to print the JSON-LD data container at all" - the conditional there does exactly this. But we can do it differently - in the (other) template.

Have other thoughts but need to shift focus for a bit.

Reply:
Sorry, yes when `$jsonld_enable` is falsey, `$json_ld_data` is not blank. What I meant to say is we could set `$json_ld_data` to a blank string.

Either way though this approach works fine, I just thought the approach of setting `$json_ld_data` to a blank string would've been cleaner from a template perspective so there isn't unneeded logic (when it can be under the hood).